client/src/components/Header.tsx

```diff
@@ -56,31 +57,38 @@ export function Header() {
             <X size={48} className="ml-auto hover:text-ruby" />
           </button>
           <ul className="h2">
-            <li>
-              <Link
-                to="/medications"
-                className="flex justify-end items-center gap-2 mb-4 transition-ease text-[2rem] hover:text-ruby"
-                onClick={handleClick}>
-                Medications
-                <Pill size={32} />
-              </Link>
-            </li>
-            <li>
-              {!user ? (
+            {user && (
+              <li className="w-full">
                 <Link
-                  to="/sign-up"
-                  onClick={handleClick}
-                  className="flex justify-end items-center gap-2 transition-ease text-[2rem] hover:text-ruby">
-                  Register / Sign in <UserRound size={32} />
+                  to="/medications"
+                  className="nav-link"
+                  onClick={handleClick}>
+                  Medications
+                  <Pill size={32} />
+                </Link>
+              </li>
+            )}
+            {user && (
+              <li className="w-full">
+                <Link to="/profile" onClick={handleClick} className="nav-link">
+                  {user.username} <UserRound size={32} />
+                </Link>
+              </li>
+            )}
```

Comment:
Since you want to render both of these if the `user` is true, why not do this in 1 conditional check?